### ISSUE TITLE
Not Working on Some Link

### ISSUE THREAD

**Issue by ghost** [chandilsachin/VideoDownloadAndPlay#4]
it is not working on some links and video will play after whole download completed in some links

**Comment by chandilsachin** [comment]
Can you provide those links so that i can reproduce issue?

**Comment by ghost** [comment]
http://www.sample-videos.com/video/mp4/240/big_buck_bunny_240p_30mb.mp4
http://myphotokeyboard.in/share_chat/upload_media/videos/1503644548.mp4
http://myphotokeyboard.in/share_chat/upload_media/videos/1503658045.mp4
http://myphotokeyboard.in/share_chat/upload_media/videos/1503659454.mp4

i have tried all these above links and download part is working well but didn't streaming while downloading

**Comment by ghost** [comment]
hey sachin have you checked those links?

**Comment by chandilsachin** [comment]
Hi dipen, i will check it tomorrow as i will get spare time on weekends. Thanks for your patience.

**Comment by ghost** [comment]
hey sachin , for all above links your processRequest Method in LocalFileStreamingServer throws exception
like "Ignoring sendto failed: ECONNRESET (Connection reset by peer)"
please give me solution for this
Thank you

**Comment by ghost** [comment]
hey sachin, have you checked above error?

**Comment by chandilsachin** [comment]
Yes i checked it and yes its not working with those links. I just got off from Vacation, so I am on my way solving them.

**Comment by ganeshpadhayay** [comment]
Hey Sachin, any updates on this issue. I am facing the same problem.

**Comment by chandilsachin** [comment]
Totally forgot it. Thanks.

Give me some time i will look into it today evening. You can also check it
meanwhile.

Best, Sachin.

**Comment by ganeshpadhayay** [comment]
I have a doubt - 
Once the video is downloaded in the local storage and then if I kill the app and restart it then it starts the downloading again but that is not required as the file is already downloaded and present in the local file system.
Is there a way through which we can host that already downloaded file on the newly created proxy server and start streaming without downloading it again?
Thanks for the quick response!

**Comment by chandilsachin** [comment]
It's actually in todo, i didn't have much time to implement it. Before
downloading, we just need to check if file is already there, if it's there
just add headers accordingly and resume download.

**Comment by ganeshpadhayay** [comment]
I tried playing with headers but no success. If you get some free time could you look into these issues? 
Thank you

**Comment by chandilsachin** [comment]
Yeah, Sure.

**Comment by ganeshpadhayay** [comment]
Hi Sachin,
Did you find any solution for those specific links where the streaming is starting after the complete download?
I have implemented that 'don't start downloading again if the file is already downloaded' part successfully.
Facing issues with those links.
http://www.sample-videos.com/video/mp4/240/big_buck_bunny_240p_30mb.mp4
http://myphotokeyboard.in/share_chat/upload_media/videos/1503644548.mp4
http://myphotokeyboard.in/share_chat/upload_media/videos/1503658045.mp4
http://myphotokeyboard.in/share_chat/upload_media/videos/1503659454.mp4

Hoping for a positive reply.
Thanks!

**Comment by chandilsachin** [comment]
I tried solving it last weekend. I am unable to figure out the problem as i don't have expertise in Video streaming. It will need more time that i am unable to spare. I am really sorry @ganeshpadhayay. If you want to take that time and solve it i can help you.

**Comment by ganeshpadhayay** [comment]
Hi Sachin,
Yes, I will surely work on this. Now I want to stream and download multiple videos simultaneously. How do I approach this? Multiple ports for multiple videos or multiple proxy servers? Is there a way to use a single server for multiple files and we can selectively start and stop the downloading of a particular file?
Thanks!

**Comment by chandilsachin** [comment]
For multiple videos you have to create a new server for each. This lib uses socket server to communicate with MediaPlayer. I assume you use recyclerView or something for multiple VideoView. In this case just keep stoping server attached to a VideoView that is no longer on Screen.

One more thing, This lib uses AsyncTask for downloading video file. You can convert it to `Foreground Service`. Let me add you to collaborator list. Thanks for taking this up.

**Comment by ganeshpadhayay** [comment]
Hi,
Thanks for adding me to the collaborator list.
For multiple servers, How do I choose IPs? I am not much into networking so Can I make random IPs or 127.0.0.1 is the only one we can choose for proxy servers?
Thanks

**Comment by ganeshpadhayay** [comment]
127 . * . * . *  can be used to create multiple proxy servers, right?

**Comment by chandilsachin** [comment]
No, You just need to change port no.
IP's would be same: 127.0.0.1.

**Comment by ganeshpadhayay** [comment]
But in that case, how do we stop a particular file download? stopping the server would stop all the downloads. Is there any way we can stop a particular process(port number refers to a particular process, right?) without stopping other ports?

**Comment by chandilsachin** [comment]
Each socket server should have a separate thread in that case this problem wont occur.
We can have multiple server in a machine with different port like : `127.0.0.1:23456`, `127.0.0.1:23457`, `127.0.0.1:23458` .... `127.0.0.1:23470`.

You can create a port pool that can generate unique ports or reuse closed server port.

**Comment by ganeshpadhayay** [comment]
Yes. Understood! 
Thanks

**Comment by ghost** [comment]
> Hi Sachin,
> Did you find any solution for those specific links where the streaming is starting after the complete download?
> I have implemented that 'don't start downloading again if the file is already downloaded' part successfully.
> Facing issues with those links.
> http://www.sample-videos.com/video/mp4/240/big_buck_bunny_240p_30mb.mp4
> http://myphotokeyboard.in/share_chat/upload_media/videos/1503644548.mp4
> http://myphotokeyboard.in/share_chat/upload_media/videos/1503658045.mp4
> http://myphotokeyboard.in/share_chat/upload_media/videos/1503659454.mp4
> 
> Hoping for a positive reply.
> Thanks!

Hey @ganeshpadhayay. Please can you share how you did the download continuation if file already exists. Thanks
